tendermint-testgen/src/commit.rs

```diff
@@ -0,0 +1,190 @@
+use gumdrop::Options;
+use serde::Deserialize;
+use simple_error::*;
+use tendermint::{block, lite};
+
+use crate::{helpers::*, Generator, Header, Validator, Vote};
+
+#[derive(Debug, Options, Deserialize, Clone)]
+pub struct Commit {
+    #[options(help = "header (required)", parse(try_from_str = "parse_as::<Header>"))]
+    pub header: Option<Header>,
+    #[options(
+        help = "votes in this commit (default: from header)",
+        parse(try_from_str = "parse_as::<Vec<Vote>>")
+    )]
+    pub votes: Option<Vec<Vote>>,
+    #[options(help = "commit round (default: 1)")]
+    pub round: Option<u64>,
+}
+
+impl Commit {
+    pub fn new(header: Header) -> Self {
+        Commit {
+            header: Some(header),
+            round: None,
+            votes: None,
+        }
+    }
+    set_option!(votes, Vec<Vote>);
+    set_option!(round, u64);
+
+    /// Generate commit votes from all validators in the header.
+    /// This function will panic if the header is not present
+    pub fn generate_default_votes(mut self) -> Self {
```

Comment:
Shouldn't this be a part of `Commit::new` ? Or is there a reason we're not adding values for votes and round there?

Reply:
I don't think so. There are two scenarios we want to cover, probably:
* all (or most) votes are present, and either there are no or very little deviation in the votes from the auto-generated default votes.  In that cause the test writer will go over the route `Commit::new()` -> `generate_default_votes()` -> `modify the votes`
* the desired votes structure differs substantially from the auto-generated one. in that case the user would go over `Commit::new()` -> `add votes manually`.

In general, as a design principle of this library, we want to leave as many parameters optional as possible, but to allow the user to set them explicitely when needed. In the cause of `round`, there is a reasonable default value, `1`, which should probably suffice in most cases, that's why we leave it optional.

Reply:
Aha, you're right. Thanks for explaining @andrey-kuprianov !

Reply:
@Shivani912 actually, you were right:) See 44c3f92.

It is a bit difficult to achieve the right balance here, because `round` and `votes` parameters of a `Commit` are not independent. Would love to hear your opinion on the new variant.